solc/CommandLineParser.cpp

```diff
@@ -0,0 +1,1243 @@
+static string const g_argGas = g_strGas;
+static string const g_argHelp = g_strHelp;
+static string const g_argImportAst = g_strImportAst;
+static string const g_argInputFile = g_strInputFile;
+static string const g_argYul = g_strYul;
+static string const g_argIR = g_strIR;
+static string const g_argIROptimized = g_strIROptimized;
+static string const g_argEwasm = g_strEwasm;
+static string const g_argExperimentalViaIR = g_strExperimentalViaIR;
+static string const g_argLibraries = g_strLibraries;
+static string const g_argLink = g_strLink;
+static string const g_argMachine = g_strMachine;
+static string const g_argMetadata = g_strMetadata;
+static string const g_argMetadataHash = g_strMetadataHash;
+static string const g_argMetadataLiteral = g_strMetadataLiteral;
+static string const g_argModelCheckerContracts = g_strModelCheckerContracts;
+static string const g_argModelCheckerEngine = g_strModelCheckerEngine;
+static string const g_argModelCheckerTargets = g_strModelCheckerTargets;
+static string const g_argModelCheckerTimeout = g_strModelCheckerTimeout;
+static string const g_argNatspecDev = g_strNatspecDev;
+static string const g_argNatspecUser = g_strNatspecUser;
+static string const g_argOpcodes = g_strOpcodes;
+static string const g_argOptimize = g_strOptimize;
+static string const g_argOptimizeRuns = g_strOptimizeRuns;
+static string const g_argOutputDir = g_strOutputDir;
+static string const g_argSignatureHashes = g_strSignatureHashes;
+static string const g_argStandardJSON = g_strStandardJSON;
+static string const g_argStorageLayout = g_strStorageLayout;
+static string const g_argStrictAssembly = g_strStrictAssembly;
+static string const g_argVersion = g_strVersion;
+static string const g_argIgnoreMissingFiles = g_strIgnoreMissingFiles;
+static string const g_argColor = g_strColor;
+static string const g_argNoColor = g_strNoColor;
+static string const g_argErrorIds = g_strErrorIds;
+
+/// Possible arguments to for --combined-json
+static set<string> const g_combinedJsonArgs
+{
+	g_strAbi,
+	g_strAsm,
+	g_strAst,
+	g_strBinary,
+	g_strBinaryRuntime,
+	g_strCompactJSON,
+	g_strFunDebug,
+	g_strFunDebugRuntime,
+	g_strGeneratedSources,
+	g_strGeneratedSourcesRuntime,
+	g_strInterface,
+	g_strMetadata,
+	g_strNatspecUser,
+	g_strNatspecDev,
+	g_strOpcodes,
+	g_strSignatureHashes,
+	g_strSrcMap,
+	g_strSrcMapRuntime,
+	g_strStorageLayout
+};
+
+/// Possible arguments to for --machine
+static set<string> const g_machineArgs
+{
+	g_strEVM,
+	g_strEwasm
+};
+
+/// Possible arguments to for --yul-dialect
+static set<string> const g_yulDialectArgs
+{
+	g_strEVM,
+	g_strEwasm
+};
+
+/// Possible arguments to for --metadata-hash
+static set<string> const g_metadataHashArgs
+{
+	g_strIPFS,
+	g_strSwarm,
+	g_strNone
+};
+
+static void version()
+{
+	sout() <<
+		"solc, the solidity compiler commandline interface" <<
+		endl <<
+		"Version: " <<
+		solidity::frontend::VersionString <<
+		endl;
+	exit(0);
```

Comment:
```suggestion
	exit(EXIT_SUCCESS);
```
(0 and != 0 meanings for process exit is AFAIK undefined, `EXIT_SUCCESS` / `EXIT_FAILURE` are part of `<cstdlib>`. 

At first I thought this function is only to construct the string, so maybe give it the name of what it does? `printVersionAndExit()`? (same below, then also mark as `[[noreturn]]`.

Reply:
I think these should not exit at all and instead just return a value so that `main()` can exit instead. For example I cannot even test `--version` and `--license` if they just exit. But this requires a bigger change in how the parsing reports the result so I decided to leave it out of the PR. This will be fixed separately.

And your suggestions are good. I'm going to apply them.